packages/main/src/plugin/container-registry.spec.ts

```
@@ -1220,6 +1220,37 @@ test('pull unknown image fails with error 403', async () => {
   );
 });
 
+test('pulling an image with platform=linux/arm64 will add the platform to the pull options', async () => {
+  const getMatchingEngineFromConnectionSpy = vi.spyOn(containerRegistry, 'getMatchingEngineFromConnection');
+
+  const pullMock = vi.fn();
+
+  const fakeDockerode = {
+    pull: pullMock,
+    modem: {
+      followProgress: vi.fn(),
+    },
+  } as unknown as Dockerode;
+
+  getMatchingEngineFromConnectionSpy.mockReturnValue(fakeDockerode);
+
+  const containerConnectionInfo = {} as ProviderContainerConnectionInfo;
+
+  pullMock.mockResolvedValue({ statusCode: 201 });
+
+  const callback = vi.fn();
+
+  // Ignore no floating promises warning as we just care about the pullMock call, not the promise
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

issue: we should not ignore this error

could we just add await before the containerRegistry.pullImage call ?

---

I thought this too, but it get's "stuck" when doing await, as if it's waiting for something to complete, despite me having the call back to resolve. 

I'll try another way of doing the test